### PR TITLE
Examples and brief docs for oci-factory's reusable workflows

### PR DESCRIPTION
**Pull request by clay-lake**
Features included in this PR:
- Introduction and brief documentation for oci-factory's reusable workflows.
- Various samples on how to integrate Build-Rock and Test-Rock workflows.

### CODE REVIEW

File: oci_factory_workflows/REAME.md
Comment (cjdcordeiro) [inline oci_factory_workflows/REAME.md:1]: this would be nice within the oci-factory repo docs

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
LGTM. Can we rename the GH workflows to have some form of identifier? Since this is a sort of catch-all repo, it's safer to name things as explicitly as possible, especially when the files may be mixed with others from different contexts.

So, what about adding a prefix like "oci_factory"? And the same for the workflow `name`s

File: .github/workflows/build_and_test_eicar_rock.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Build and Test EICAR Rock
```
Comment (clay-lake) [inline .github/workflows/build_and_test_eicar_rock.yaml:1]:
```suggestion
# Warning: This workflow is designed to fail. When merging in a PR, pushing an empty commit will bypass checks and allow the PR to be merged.
name: Build and Test EICAR Rock
```

File: .github/workflows/test_external_rock.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Test External Rock
```
Comment (clay-lake) [inline .github/workflows/test_external_rock.yaml:1]:
```suggestion
# Warning: This workflow is designed to fail. When merging in a PR, pushing an empty commit will bypass checks and allow the PR to be merged.
name: Test External Rock
```

**Review by cjdcordeiro — APPROVED** [review]
great thanks

**Comment by cjdcordeiro** [comment]
Can you squash and sign? The branch protection doesn't allow me to merge without signing the commits

**Review by zhijie-yang — APPROVED** [review]
LGTM, thanks.

**Pull request merged**